tracing-core/src/metadata.rs

```diff
@@ -208,12 +210,69 @@ impl<'a> fmt::Debug for Metadata<'a> {
         let mut meta = f.debug_struct("Metadata");
         meta.field("name", &self.name)
             .field("target", &self.target)
-            .field("level", &self.level);
+            .field("level", &self.level)
+            .field("location", &self.location)
+            .field("fields", &format_args!("{}", self.fields))
+            .field("callsite", &self.callsite())
+            .field("kind", &self.kind)
+            .finish()
+    }
+}
+
+// ===== impl Location =====
+impl<'a> Location<'a> {
+    /// Construct new source code location where the `Metadata` originated from with a file, line, column, module_path
+    pub const fn new(
+        file: Option<&'a str>,
+        line: Option<u32>,
+        column: Option<u32>,
+        module_path: Option<&'a str>,
+    ) -> Self {
+        Self {
+            file,
+            line,
+            column,
+            module_path,
+        }
+    }
+
+    /// Returns the column number in the source code file where this
+    /// `Metadata` originated from, or `None` if the column number is unknown.
+    pub fn column(&self) -> Option<u32> {
+        self.column
+    }
+
+    /// Returns the path to the Rust module where this
+    /// `Metadata` originated from, or `None` if the module path is unknown.
+    pub fn module_path(&self) -> Option<&'a str> {
+        self.module_path
+    }
+
+    /// Returns the name of the source code file where this
+    /// `Metadata` originated from, or `None` if the file is unknown
+    pub fn file(&self) -> Option<&'a str> {
+        self.file
+    }
+
+    /// Returns the line number in the source code file where this
+    /// `Metadata` originated from, or `None` if the line number is unknown.
+    pub fn line(&self) -> Option<u32> {
+        self.line
+    }
+}
+
+impl<'a> fmt::Debug for Location<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut meta = f.debug_struct("Location");
```

Comment:
nit: since this is the `Debug` impl for the `Location` type, not the `Metadata` type, i might call this variable something different...